mongoimport/mongoimport_test.go

```diff
@@ -31,29 +33,36 @@ const (
 
 // checkOnlyHasDocuments returns an error if the documents in the test
 // collection don't exactly match those that are passed in
-func checkOnlyHasDocuments(sessionProvider db.SessionProvider, expectedDocuments []bson.M) error {
+func checkOnlyHasDocuments(sessionProvider db.SessionProvider, expectedDocuments []gbson.M) error {
 	session, err := sessionProvider.GetSession()
 	if err != nil {
 		return err
 	}
-	defer session.Close()
 
-	collection := session.DB(testDb).C(testCollection)
-	dbDocuments := []bson.M{}
-	err = collection.Find(nil).Sort("_id").All(&dbDocuments)
+	collection := session.Database(testDb).Collection(testCollection)
+	cursor, err := collection.Find(nil, gbson.D{}, driverOpts.Find().SetSort(gbson.D{{"_id", 1}}))
 	if err != nil {
 		return err
 	}
-	if len(dbDocuments) != len(expectedDocuments) {
-		return fmt.Errorf("document count mismatch: expected %#v, got %#v",
-			len(expectedDocuments), len(dbDocuments))
-	}
-	for index := range dbDocuments {
-		if !reflect.DeepEqual(dbDocuments[index], expectedDocuments[index]) {
+
+	var numDocs int
+	for cursor.Next(nil) {
+		var doc gbson.M
+		if err = cursor.Decode(&doc); err != nil {
+			return err
+		}
+
+		if !reflect.DeepEqual(doc, expectedDocuments[numDocs]) {
 			return fmt.Errorf("document mismatch: expected %#v, got %#v",
-				expectedDocuments[index], dbDocuments[index])
+				expectedDocuments[numDocs], doc)
```

Comment:
This will panic if numDocs is out of range.  (Note how the original accumulated via `All` first to confirm length.)

Reply:
Reordered to accumulate all of the docs into a slice first.